### PR TITLE
Update match-expressions.md

### PR DESCRIPTION
**Pull request by cchiw**
#274

### CODE REVIEW

File: src/control-flow/match-expressions.md [hunk 1]
```diff
@@ -21,3 +21,12 @@ expression of the block, if any. In the example above, the type is `()`.
 
 See [pattern matching](../pattern-matching.md) for more details on patterns in
 Rust.
+
+<details>
+    
+* Save the match expression to a variable and print it out.
+* Remove `.as_deref()` and explain the error.
+    * `std::env::Agrs().next` returns an option borrowed string.
```
Comment (mgeisler) [inline src/control-flow/match-expressions.md:29]:
```suggestion
    * `std::env::Agrs().next` returns an `Option<&String>`, but we cannot match against `String`.
```

File: src/control-flow/match-expressions.md [hunk 2]
```diff
@@ -21,3 +21,12 @@ expression of the block, if any. In the example above, the type is `()`.
 
 See [pattern matching](../pattern-matching.md) for more details on patterns in
 Rust.
+
+<details>
+    
+* Save the match expression to a variable and print it out.
+* Remove `.as_deref()` and explain the error.
+    * `std::env::Agrs().next` returns an option borrowed string. 
+    * `as_deref()` takes an `option` type and calls the deref on the inside value in the case of a `Some` variant.
```
Comment (mgeisler) [inline src/control-flow/match-expressions.md:30]:
```suggestion
    * `as_deref()` transforms an `Option<T>` to `Option<T::Target>`. In our case, this turns `Option<&String>` into `Option<&str>`.
```

File: src/control-flow/match-expressions.md [hunk 3]
```diff
@@ -21,3 +21,12 @@ expression of the block, if any. In the example above, the type is `()`.
...
+    * By using `as_deref()`, we are able to match against a String literal.
```
Comment (mgeisler) [inline src/control-flow/match-expressions.md:31]:
```suggestion
    * We can now use pattern matching to match against the `&str` inside `Option`.
```

**Review by mgeisler — APPROVED** [review]

File: src/control-flow/match-expressions.md [hunk 4]
```diff
@@ -21,3 +21,12 @@ expression of the block, if any. In the example above, the type is `()`.
 
 See [pattern matching](../pattern-matching.md) for more details on patterns in
 Rust.
+
+<details>
+    
+* Save the match expression to a variable and print it out.
+* Remove `.as_deref()` and explain the error.
+    * `std::env::Agrs().next` returns an `Option<&String>`, but we cannot match against `String`.
```
Comment (qwandor) [inline src/control-flow/match-expressions.md:29]: I guess you mean `Args()` and `next()`.

Comment (qwandor) [reply src/control-flow/match-expressions.md:29]: Or rather, `args().next()`, case is important.

Comment (cchiw) [reply src/control-flow/match-expressions.md:29]: Follow-up in #400

**Pull request merged**